spring-boot/admin/src/main/java/eu/coatrack/admin/service/GatewayConfigFilesStorage.java

```diff
@@ -0,0 +1,104 @@
+package eu.coatrack.admin.service;
+
+/*-
+ * #%L
+ * coatrack-admin
+ * %%
+ * Copyright (C) 2013 - 2020 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import eu.coatrack.api.Proxy;
+import eu.coatrack.api.ServiceApi;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+
+/**
+ *
+ * @author perezdf, ChristophBaier
+ */
+
+@Slf4j
+@Component
+public class GatewayConfigFilesStorage {
+
+    @Value("${proxy.config.files.folder}")
+    private String gatewayConfigFilesFolderLocation;
+
+    @Value("${ygg.admin.api-base-url-for-gateway}")
+    private String adminApiBaseUrlForGateway;
+
+    @Value("${is-run-via-docker-compose-setup}")
+    private boolean isDockerComposeSetup;
+
+    @PostConstruct
+    private void emptyProxyConfigFilesFolderIfExistsForNonDockerComposeDeployments() throws IOException {
+        if (!isDockerComposeSetup){
+            Path targetFolderPath = Paths.get(System.getProperty("user.dir") + "/target");
+            if (Files.notExists(targetFolderPath))
+                Files.createDirectory(targetFolderPath);
+
+            Path gatewayConfigFilesFolderPath = Paths.get(gatewayConfigFilesFolderLocation);
+            if (Files.exists(gatewayConfigFilesFolderPath)){
+                FileUtils.deleteDirectory(gatewayConfigFilesFolderPath.toFile());
```

Comment:
If I understand correctly, CoatRack deletes something from my hard drive on every start-up. Does not seem like a good idea to me, as it could in some situations delete important stuff by accident. To be discussed.

Reply:
It is okay for an application to create files/folders on start-up, but most people would not expect it to delete files. Will it lead to problems if we keep that directory?

Reply:
This directory is ignored by git. It contains only Gateway Config Files created when running CoatRack via CLI/IDE without using docker which is only performed by developers. Therefore, to keep this directory small and clean, it is emptied on every startup. 

This cleanup will not affect the persistence of Gateway Config File docker volume. I don't see any problems here, yet.

Reply:
If this is only to be used during development, then I would suggest to remove it from the code and move it to some script that developers use to run the application locally (or to move it to a cleanup script that developers can optionally use).

In doubt we should have another discussion about this. The main problem I see here is: if I understand correctly, starting up the application will delete stuff from a _previous_ run of the application, which is not at all relevant for the current run. This should imho not happen, as maybe for some reason (e.g. tests) developers would want to keep files from the previous run (and usually you do not expect that a new run will delete stuff from a previous run, except maybe for log files).

Reply:
I removed the folder deletion logic. 

Actually, you don't need a cleanup script. You can just delete the folder. On the next run of your application, a new and empty folder is created.